Added features to Concord232 Alarm Panel

AZDane opened the pull request:
## Description: Adds functionality to Concord232 Alarm Panel. 2 additional attributes can be defined in configuration.yaml. 'code' which allows for defining a code and 'mode' which allows for silent activation of the ARM HOME option

Currently Concord232 Server can Arm/Disarm without providing code or by providing wrong code. Updated code adds functionality to define a code in configuration.yaml.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#9039

## Example entry for `configuration.yaml` (if applicable):
```yaml
alarm_control_panel:
  - platform: concord232
     host: 192.168.1.172
     code: 1234
     mode: silent
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

AZDane committed:
Update alarm_control_panel.py

Added ability to define code and mode in configuration.yaml
code
  (string)(Optional) If defined, specifies a code to enable or disable the alarm in the frontend. Only one of code and code_template can 
   be specified.
mode
  (string)(Optional) audible/silent if defined, specifies wether Alarm Panel should be audible or silent when armed in Home Mode.
  Default Value: audible

AZDane committed:
Expanded concord232 Alarm Panel Features.

Currently concord232 arms and disarms panel regardless of code entry (none/random)
Changes ads ability to define code in configuration.yaml. Additionally ability to arm panel silent has been added.
code
  (string)(Optional) If defined, specifies a code to enable or disable the alarm in the frontend. 
mode
  (string)(Optional) audible/silent if defined, specifies wether Alarm Panel should be audible or silent when armed in Home Mode.
  Default Value: audible

AZDane committed:
Update alarm_control_panel.py

Deleted Trailing White Space

homeassistant commented:
Hi @AZDane, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22411).

Once you do that we will be able to review and accept this pull request.

Thanks!

AZDane commented:
Documentation added/updated in home-assistant.io#9039

AZDane committed:
Updated concord232 binary sensor

wrong indention of add_entities(sensors, True) caused same sensor to be added multiple times since it was placed inside loop

AZDane commented:
Added concord binary sensor to the request since it contains an error causing same sensor to be added multiple times

AZDane committed:
Update alarm_control_panel.py

Further simplification. Ok. I will stop now :)

awarecan commented:
Please rebase your branch to get our new CI configuration to your branch, or create a new branch base on current `dev` branch. I will recommend create a new branch, reimplement your changes, then create a new pull request. Because you said you are not familiar with git, and this one has not been reviewed yet.

If you choose rebase, following are some basic git command
```bash
# in your branch do
git remote add upstream https://github.com/home-assistant/home-assistant.git
git fetch upstream
git rebase upstream/dev
git push -f
```
